src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs

```diff
@@ -245,7 +283,24 @@ private async Task DoSend()
                 if (!buffer.IsEmpty)
                 {
                     _sender = _socketSenderPool.Rent();
-                    await _sender.SendAsync(_socket, buffer);
+                    var transferResult = await _sender.SendAsync(_socket, buffer);
+
+                    if (transferResult.HasError && IsConnectionResetError(transferResult.SocketError!.SocketErrorCode))
+                    {
+                        var ex = transferResult.SocketError!;
+                        shutdownReason = new ConnectionResetException(ex.Message, ex);
+                        SocketsLog.ConnectionReset(_logger, this);
+
+                        break;
+                    }
+
+                    if (transferResult.HasError && IsConnectionAbortError(transferResult.SocketError!.SocketErrorCode))
+                    {
+                        shutdownReason = transferResult.SocketError!;
+
+                        break;
+                    }
+
```

Comment:
If we've gotten to this point and `transferResult.HasError`, we should be doing `unexpectedError = shutdownReason = transferResult.SocketError`.

Reply:
I did the change, but I don't think it was the behavior before.